tests/integration/rust/src/main.rs

```diff
@@ -0,0 +1,108 @@
+use arrow::csv::ReaderBuilder;
+use arrow::datatypes::{DataType, Field, Schema};
+use arrow::record_batch::RecordBatch;
+use clap::Parser;
+use icelake::transaction::Transaction;
+use icelake::Table;
+use opendal::services::S3;
+use opendal::Operator;
+use std::fs::File;
+use std::sync::Arc;
+
+#[derive(Parser, Debug)]
```

Comment:
I prefer using `dotenvy` and `libtest-mimic` for smoother integration with `cargo test`. It is highly likely that we need to test against various storage services and catalogs.

cc opendal's for example: https://github.com/apache/incubator-opendal/blob/main/core/tests/behavior/main.rs